cg/meta/store/base.py

```diff
@@ -29,26 +35,110 @@ def add_new_analysis(bundle_data, case_obj, status, version_obj):
 
 def reset_case_action(case_obj):
     """ Resets action on case """
+
     case_obj.action = None
 
 
-def get_files(deliverables: dict, pipeline: str) -> dict:
+def get_files(deliverables: dict, pipeline: list) -> list:
     """Get all deliverable files from the pipeline"""
 
-    data = [
+    files = _get_files_non_index(deliverables, pipeline)
+    index_files = _get_files_index(deliverables, pipeline)
+
+    files.extend(index_files)
+
+    _check_mandatory_tags(files)
+    _convert_tags(files)
+
+    return files
+
+
+def _get_files_non_index(deliverables: dict, pipeline: list) -> list:
+    """ Get all files that are not index files from the deliverables file """
+
+    return [
         {"path": file["path"], "tags": get_tags(file, pipeline), "archive": False}
         for file in deliverables["files"]
     ]
 
-    return data
 
+def _get_files_index(deliverables: dict, pipeline: list) -> list:
+    """ Get all index files from the deliverables file """
 
-def get_tags(file: dict, pipeline: str) -> list:
+    return [
+        {"path": file["path_index"], "tags": get_tags(file, pipeline), "archive": False,}
+        for file in deliverables["files"]
+        if file["path_index"]
+    ]
+
+
+def get_tags(file: dict, pipeline_tags: list) -> list:
     """Get all tags for a file"""
 
-    all_tags = [file["format"], file["id"], file["step"], file["tag"], pipeline]
+    all_tags = [file["format"], file["id"], file["step"], file["tag"]]
+    all_tags.extend(pipeline_tags)
     unique_tags = set(all_tags)
     only_existing_tags = unique_tags - set([None])
     sorted_tags = sorted(list(only_existing_tags))
 
     return sorted_tags
+
+
+def build_bundle(config_data: dict, analysisinfo_data: dict, deliverables: dict) -> dict:
+    """Create a new bundle to store in Housekeeper"""
+
+    pipeline = config_data["samples"][0]["type"]
+    pipeline_tag = HK_TAGS[pipeline]
+
+    data = {
+        "name": config_data["case"],
+        "created": analysisinfo_data["date"],
+        "pipeline_version": analysisinfo_data["version"],
+        "files": get_files(deliverables, pipeline_tag),
+    }
+    return data
+
+
+def _convert_tags(data: list):
```

Comment:
This one should import a set of PIPELINE_TAGS instead to make it agnostic to pipeline